metals/src/main/scala/scala/meta/internal/metals/ScalafmtConfig.scala

```diff
@@ -0,0 +1,206 @@
+      includeFilters: List[PathMatcher],
+      excludeFilters: List[PathMatcher]
+  ) {
+
+    def overrideFor(path: AbsolutePath): Option[ScalafmtDialect] = {
+      val fs = FileSystems.getDefault
+      fileOverrides
+        .find { case (pm, _) =>
+          val matcher = fs.getPathMatcher(pm)
+          matcher.matches(path.toNIO)
+        }
+        .map(_._2)
+    }
+
+    def isExcluded(path: AbsolutePath): Boolean = {
+      excludeFilters.exists(_.matches(path)) && !includeFilters.exists(
+        _.matches(path)
+      )
+    }
+  }
+
+  def update(
+      configText: String,
+      version: Option[String],
+      runnerDialect: Option[ScalafmtDialect]
+  ): String = {
+
+    def docFrom(s: String): ConfigDocument = {
+      val options = ConfigParseOptions.defaults().setSyntax(ConfigSyntax.CONF)
+      ConfigDocumentFactory.parseString(s, options)
+    }
+
+    def withUpdatedVersion(content: String, v: String): String = {
+      val doc = docFrom(content)
+      if (doc.hasPath("version"))
+        doc.withValueText("version", '"' + v + '"').render
+      else {
+        // prepend to the beggining of file
+        val sb = new StringBuilder
+        sb.append(s"""version = "$v"""")
+        sb.append(System.lineSeparator)
+        sb.append(content)
+        sb.toString
+      }
+    }
+
+    def withUpdatedDialect(content: String, d: ScalafmtDialect): String = {
+      val doc = docFrom(content)
+      if (doc.hasPath("runner.dialect"))
+        doc.withValueText("runner.dialect", d.value).render
+      else {
+        // append to the end
+        val sb = new StringBuilder
+        sb.append(content)
+        val sep = System.lineSeparator
+        val lastLn = content.endsWith(sep)
+        if (!lastLn) sb.append(sep)
+        sb.append(s"runner.dialect = ${d.value}")
+        sb.append(sep)
+        sb.toString
+      }
+    }
+
+    val applyVersion =
+      version.fold(configText)(withUpdatedVersion(configText, _))
+    runnerDialect.fold(applyVersion)(withUpdatedDialect(applyVersion, _))
+  }
+
+  def parse(path: AbsolutePath): Try[ConfigValues] =
+    Try(ConfigFactory.parseFile(path.toFile)).flatMap(parse)
+
+  def parse(text: String): Try[ConfigValues] =
+    Try(ConfigFactory.parseString(text)).flatMap(parse)
+
+  def parse(config: Config): Try[ConfigValues] = {
+
+    def getVersion(conf: Config): Try[Option[String]] =
+      Try {
+        if (conf.hasPath("version")) Some(conf.getString("version"))
+        else None
+      }
+
+    def getRunnerDialectRaw(conf: Config): Option[ScalafmtDialect] = {
+      if (conf.hasPath("runner.dialect")) {
+        val v = conf.getString("runner.dialect")
+        ScalafmtDialect.fromString(v)
+      } else
+        None
+    }
+
+    def getRunnerDialect(
+        conf: Config
+    ): Try[Option[ScalafmtDialect]] = {
+      Try(getRunnerDialectRaw(conf))
+    }
+
+    def getFileOverrides(
+        conf: Config
+    ): Try[Map[String, ScalafmtDialect]] = {
```

Comment:
```suggestion
    ): Map[String, ScalafmtDialect] = {
```
and just default to empty map?